File: Lecture08/InterFragmentCommunication/app/src/main/java/com/codingblocks/interfragmentcommunication/MainActivity.java
package com.codingblocks.interfragmentcommunication;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements ClickHandler {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getSupportFragmentManager().beginTransaction()
                .replace(R.id.container,
                        new NoteFragment())
                .commit();
    }

    @Override
    public void onRvItemClicked(Note currentNote) {
        Intent intent = new Intent(this, NewActivity.class);
        intent.putExtra("TITLE", currentNote.getTitle());
        intent.putExtra("SUBTITLE", currentNote.getSubtitle());

        startActivity(intent);
    }

}

File: Lecture20/BroadCasts/app/src/main/java/com/codingblocks/broadcasts/MainActivity.kt
package com.codingblocks.broadcasts

import android.content.Intent
import android.content.IntentFilter
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.content.LocalBroadcastManager

class MainActivity : AppCompatActivity() {

    private val myReceiver = MyReceiver()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val intent = Intent().apply {
            action = "mycustombroadcast"
            putExtra("USERNAME", "test")
            putExtra("PASSWORD", "password")
        }


        //Ensures that the intent never leaks out of your app
        LocalBroadcastManager.getInstance(this).sendBroadcast(intent)

//        sendBroadcast(intent)
    }

    override fun onStart() {
        super.onStart()
        val intentFilter = IntentFilter().apply {
            //            addAction(Intent.ACTION_POWER_CONNECTED)
//            addAction(Intent.ACTION_POWER_DISCONNECTED)
//            addAction(Intent.ACTION_AIRPLANE_MODE_CHANGED)
            addAction("mycustombroadcast")
        }

        LocalBroadcastManager.getInstance(this).registerReceiver(myReceiver, intentFilter)

//        registerReceiver(myReceiver, intentFilter)
    }


    override fun onStop() {
        super.onStop()
        LocalBroadcastManager.getInstance(this).unregisterReceiver(myReceiver)
//        unregisterReceiver(myReceiver)
    }

}
